File: reverse_bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        String s=Integer.toBinaryString(n);
        String t="";
        for(int i=0;i<32-s.length();i++){
            t+='0';
        }
        s=t+s;
        t="";
        for(int i=31;i>=0;i--){
            t+=s.charAt(i);
        }
        return Integer.parseInt(t,2);
        
    }
}

File: reverse_bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        string s="";
        for(int i=0;i<32;i++){
            int temp=n%2;
            n/=2;
            s+=(char)temp;
        }
        uint32_t res;
        for(int i=0;i<32;i++){
            res=res*10+int(s[i]);
        }
        return res;
    }
};